drivers/sensor/tdk/icp201xx/icp201xx_drv.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_ICP201XXX_ICP101XXX_H_
+#define ZEPHYR_DRIVERS_SENSOR_ICP201XXX_ICP101XXX_H_
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/sensor.h>
+#include "Icp201xx.h"
+#include "Icp201xxSerif.h"
+
+union icp201xx_bus {
+#if CONFIG_SPI
+	struct spi_dt_spec spi;
+#endif
+#if CONFIG_I2C
+	struct i2c_dt_spec i2c;
+#endif
+};
+
+typedef int (*icp201xx_reg_read_fn)(const union icp201xx_bus *bus, uint8_t reg, uint8_t *buf,
+				    uint32_t size);
+typedef int (*icp201xx_reg_write_fn)(const union icp201xx_bus *bus, uint8_t reg, uint8_t *buf,
+				     uint32_t size);
+
+struct icp201xx_bus_io {
+	icp201xx_reg_read_fn read;
+	icp201xx_reg_write_fn write;
+};
+
+#if CONFIG_SPI
```

Comment:
Same here, don't wrap with `#if`, if there are no references to the variable it'll just get stripped.

Reply:
Ping